sktime/forecasting/conditional_invertible_neural_network.py

```diff
@@ -217,12 +229,20 @@ def _fit(self, y, fh, X=None):
             truncate="fill",
         ).fit_transform(y)
 
-        self.function = CurveFitForecaster(
-            self._f_statistic,
-            {"p0": self._init_param_f_statistic},
-            normalise_index=True,
-        )
-        self.function.fit(rolling_mean.dropna())
+        # Curve fit with error handling
+        try:
+            self.function = CurveFitForecaster(
+                self._f_statistic,
+                {"p0": self._init_param_f_statistic},
+                normalise_index=True,
+            )
+            self.function.fit(rolling_mean.dropna())
+        except Exception as e:
+            raise RuntimeError(
+                "Curve fitting error Please check the parameters and try again. "
```

Comment:
You might add information which parameters you are referring to. I suppose the only relevant are `_f_statistic` and `_init_param_f_statistic` or?

Reply:
i also think so , i will add the information